tests/features/glossary/glossary.feature

```diff
@@ -243,3 +243,80 @@ Feature: As a moderator or group facilitator I want to be able to add, edit and
+    Given users:
+      | Username | Roles     |
+      | regular  |           |
+      | fac      |           |
+      | mod      | moderator |
+      | fac1     |           |
+    And the following collections:
+      | title                    | description                                                                       | state     |
+      | Collection With Glossary | The Battle of Evermore of the battle and the BATTLE. Call it everMore or EVERmore | validated |
+      | The Other Collection     | desc                                                                              | validated |
+    And the following collection user membership:
+      | collection               | user | roles       |
+      | Collection With Glossary | fac  | facilitator |
+      | The Other Collection     | fac1 | facilitator |
+    And glossary content:
+      | title  | abbreviation | definition | collection               |
+      | battle | evermore     | def        | Collection With Glossary |
+
+    Given I am an anonymous user
+    When I go to the "Collection With Glossary" collection
+    Then I should not see the link "Settings" in the "Entity actions" region
+    When I click "Glossary"
+    Then I should not see the link "Edit Glossary settings"
+
+    Given I am logged in as regular
+    When I go to the "Collection With Glossary" collection
+    Then I should not see the link "Settings" in the "Entity actions" region
+    When I click "Glossary"
+    Then I should not see the link "Edit Glossary settings"
+
+    Given I am logged in as fac1
+    When I go to the "Collection With Glossary" collection
+    Then I should not see the link "Settings" in the "Entity actions" region
+    When I click "Glossary"
+    Then I should not see the link "Edit Glossary settings"
+
+    Given I am logged in as mod
+    When I go to the "Collection With Glossary" collection
+    Then I should see the link "Settings" in the "Entity actions" region
+    When I click "Glossary"
+    Then I should see the link "Edit Glossary settings"
+
+    Given I am logged in as fac
+    When I go to the "Collection With Glossary" collection
+    And I click "Settings" in the "Entity actions" region
+    Then I should see the heading "Collection With Glossary collection settings"
+    And the "Highlight only the first term appearance" checkbox should be checked
+
+    When I go to the "Collection With Glossary" collection
+
+    # Only the first occurrence should be highlighted.
+    And I should see the link "Battle"
+    And I should see the link "Evermore"
+    But I should not see the link "battle"
+    And I should not see the link "BATTLE"
+    And I should not see the link "everMore"
+    And I should not see the link "EVERmore"
+
+    When I click "Settings" in the "Entity actions" region
+    And I uncheck the box "Highlight only the first term appearance"
+    And I press "Save"
+    And I go to the "Collection With Glossary" collection
```

Comment:
After submitting the form it would be better to redirect the user automatically back to the collection homepage. The user can be a bit lost on how to return, and the back button shows the page from the browser cache, which doesn't have any changes applied.

Reply:
I fully agree.